lib/convection/control/cloud.rb

```diff
@@ -44,13 +66,14 @@ def converge(to_stack, &block)
         end
       end
 
-      def diff(to_stack, &block)
-        @cloudfile.deck.each do |stack|
-          block.call(Model::Event.new(:compare, "Compare local state of stack #{ stack.name } (#{ stack.cloud_name }) with remote template", :info))
+      def diff(to_stack, options = {}, &block)
+        filter_deck(options, &block).each do |stack|
+          #pp stack
+          block.call(Model::Event.new(:compare, "Compare local state of stack #{ stack[1].name } (#{ stack[1].cloud_name }) with remote template", :info))
 
-          difference = stack.diff
+          difference = stack[1].diff
```

Comment:
Rather than using `[1]`, how about adding `do |stack_name, stack|` to the block parameters above?